extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -749,48 +755,46 @@ class LifecycleTask extends BackbeatTask {
             new Date(obj.LastModified));
 
         if (rules.Expiration.Date &&
-        rules.Expiration.Date < Date.now()) {
+            rules.Expiration.Date < Date.now()) {
             // expiration date passed for this object
-            const entry = {
-                action: 'deleteObject',
-                target: {
-                    owner: bucketData.target.owner,
-                    bucket: bucketData.target.bucket,
-                    key: obj.Key,
-                },
-                details: {
-                    lastModified: obj.LastModified,
-                },
-            };
-            this._sendObjectEntry(entry, err => {
+            const entry = ActionQueueEntry.create('deleteObject')
+                  .addContext({
+                      origin: 'lifecycle',
+                      ruleType: 'expiration',
+                      reqId: log.getSerializedUids(),
+                  })
+                  .setAttribute('target.owner', bucketData.target.owner)
+                  .setAttribute('target.bucket', bucketData.target.bucket)
+                  .setAttribute('target.key', obj.Key)
+                  .setAttribute('details.lastModified', obj.LastModified);
+            this._sendObjectAction(entry, err => {
                 if (!err) {
-                    log.debug('sent object entry for consumption', {
-                        method: 'LifecycleTask._compareObject',
-                        entry,
-                    });
+                    log.debug(
+                        'sent object entry for consumption', Object.assign({
+                            method: 'LifecycleTask._compareObject',
```

Comment:
oops, sorry I probably forgot to change this before: ~~_compareObject~~  `_checkAndApplyExpirationRule`